From e7cee8dce3c16f5c36d735dc544999555d2cab9e Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 1 Aug 2024 13:24:17 +0200
Subject: [PATCH 1126/1695] dts: stm32: update mlahb dma range for the
 stm32mp15 series.

The  MCU SRAMs base address 0x10000000, the address 0x30000000
is an alias.
The firewall only supports the 0x10000000 base address for the ETZPC
memory protection.
update the mlahb dma-range to transform Cortex-M address to this bus
address.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 99188338e0bb773b87cceb980c0907fef17b8d3d dts: stm32: update mlahb dma range for the stm32mp15 series.]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Ib0633f2339b499b19f574bfea3b7012c074ada49
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/396466
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index 1fc3d973a..294fcd61a 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -2065,7 +2065,7 @@
 		ranges;
 		dma-ranges = <0x00000000 0x38000000 0x10000>,
 			     <0x10000000 0x10000000 0x60000>,
-			     <0x30000000 0x30000000 0x60000>;
+			     <0x30000000 0x10000000 0x60000>;
 
 		m4_rproc: m4@10000000 {
 			compatible = "st,stm32mp1-m4-tee";
-- 
2.47.2

